File: STACK/stack_using_linkedlist.c
#include <stdio.h>
#include <stdlib.h>
#define max 100

struct node
{
    int data;
    struct node *next;
};

struct node *head=NULL;
struct node *top = NULL;

void push(int count)
{
    struct node *newnode = (struct node*)malloc(sizeof(struct node));
    printf("Enter data to be Entered : ");
    scanf("%d", &newnode->data);
    newnode->next = NULL;

    if (top == NULL)
    {
        top = newnode;
        head = newnode;
    }

    else if(count==max)
    {
        printf("Stack Full");
    }
    else
    {
        newnode->next = top;
        top = newnode;
    }
}

void pop()
{
    if (top == NULL)
    {
        printf("Empty Stack");
    }
    else
    {
        struct node *temp = (struct node*)malloc(sizeof(struct node));
        temp = top;
        printf("%d", temp->data);
        top = temp->next;
        printf("\n");
    }
}

void peek()
{
    printf("Top Element: %d\n",top->data);
}


int main()
{
    int n;
    int count=0;
    printf("1. Push\n");
    printf("2. Pop\n");
    printf("3. Peek\n");
    printf("4. Exit\n");
    
    scanf("%d",&n);
    do
    {
        
        switch (n)
        {
        case 1:
        count++;
        push(count);
        break;
        
        case 2:
        pop();
        break;

        case 3:
        peek();
        break;
        
        default:
        break;
        }
        printf("\nEnter 4 to exit or Choose operation : ");
        scanf("%d",&n);
    } while (n!=4);
    
    return 0;
}

File: Python/stack_using_linkedlist.py
class node:
    def __init__(self,data):
        self.data=data
        self.link=None
class stack:
    def __init__(self):
        self.start=None
    def push(self,value):
        newnode=node(value)
        if(self.start==None):#if the stack is empty
            self.start=newnode
        else:
            temp=self.start
            while(temp.link!=None):
                temp=temp.link
            temp.link=newnode
            
    def pop(self):
        
        if(self.start==None):
            print("List is empty")
            
        elif(self.start.link==None):
            self.start=None
            
        else:
            temp=self.start
            while(temp.link.link!=None):
                temp=temp.link
            temp.link=None
            
   
    def display(self):
        if (self.start==None):
            print("list is empty")
        else:
            temp=self.start
            while(temp!=None):
                print(temp.data, end = " ")
                temp=temp.link
                
s=stack()
print("Initially: ", end=" ")
s.display()
print()
print(" After pushing 10, Stack: ", end=" ")
s.push(10)
s.display()
